src/pat/controlpanels/schemaeditor--implementation.js

```diff
@@ -25,14 +25,20 @@ export default class SchemaEditor {
     init_add_field(fieldset_id) {
         if (fieldset_id) {
             var $add_field = $("a#add-field", this.el);
-            var href = $add_field.attr("href").split("?")[0]; // get base href without any previously set ``fieldset_id``.
-            $add_field.attr("href", href + "?fieldset_id=" + fieldset_id);
+            var $fieldset = $("#form fieldset#fieldset-"+fieldset_id, this.el);
+            if ($fieldset.data('can-add-fields') === false){
+                $add_field.hide();
```

Comment:
I've tested this together with https://github.com/plone/plone.schemaeditor/pull/92 and it works like a charm. I have one suggestion: I would use the css class `disabled` for the `Add field` button when its prevented from adding instead of removing it. This also prevents clicking the button.

Reply:
@petschki Much better, indeed :)